File: application/src/main/java/com/example/server/repository/FromSizePageRequest.java
package com.example.server.repository;

import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;

public class FromSizePageRequest extends PageRequest {

    private FromSizePageRequest(Integer page, Integer size, Sort sort) {
        super(page, size, sort);
    }

    public static FromSizePageRequest of(Integer from, Integer size) {
        return of(from, size, Sort.unsorted());
    }

    public static FromSizePageRequest of(Integer from, Integer size, Sort sort) {
        return new FromSizePageRequest(from / size, size, sort);
    }
}

File: application/src/main/kotlin/com/example/server/repository/FromSizePageRequest.kt
package com.example.server.repository

import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort

class FromSizePageRequest private constructor(page: Int, size: Int, sort: Sort) :
    PageRequest(page, size, sort) {
    companion object {
        @JvmStatic
        @JvmOverloads
        fun of(from: Int, size: Int, sort: Sort = Sort.unsorted()): FromSizePageRequest {
            return FromSizePageRequest(from / size, size, sort)
        }
    }
}
